### DISCUSSION TITLE
Upgrade Hugo and Docsy

### DISCUSSION THREAD

**Discussion by Kalaiselvi84** [google/docsy#1676] [Q&A] [answered]
Team,

I am trying to upgrade Hugo(0.117.0) and Docsy(0.7.1) and getting below error,

```
Error: error building site: TOCSS: failed to transform "scss/main.scss" (text/x-scss): "/go/src/agones.dev/agones/site/themes/docsy/assets/scss/support/_bootstrap_vers_test.scss:3:3": 
  Docsy requires Bootstrap v5, but the v5 variable
  $enable-important-utilities isn't defined.
  Did you forget to update Docsy dependencies?
How are we ensuring that the dependency for the Bootstrap to v5 is available?
```

PR - https://github.com/googleforgames/agones/pull/3351

Build Log - https://gist.github.com/Kalaiselvi84/013aa9abc8ca883fb4172df7577dab63

Could you please help me fix this issue?

Thanks!

**Comment by deining** [comment]
> Team,
> 
> I am trying to upgrade Hugo(0.117.0) and Docsy(0.7.1) and getting below error,
> 
> Could you please help me fix this issue?

I was able to convert your repo and get it up and running again. See my [PR](https://github.com/googleforgames/agones/pull/3354) on your repo. I could successfully run your site locally. Some conversion work needs to be still done, see this [blog post](https://www.docsy.dev/blog/2023/docsy-0.7/) for instructions on this.

**Reply by Kalaiselvi84** [reply]
Hi @deining,

I appreciate your support!

I've submitted a [PR](https://github.com/googleforgames/agones/pull/3351) where I integrated the docsy directory directly into our existing one, as we're not keen on using `git submodules`. As a result, over 200 files were altered, as evident in the PR. However, I noticed that the [superseding PR](https://github.com/googleforgames/agones/pull/3354) which you have created has over 5000 modified files. Could you share the approach you used?

Thanks!

**Reply by deining** [reply]
When using docsy as hugo module, there is no need to keep the locally stored clone of the docsy theme. Therefore I removed the whole directory `themes/docsy`:

```
git rm themes/docsy
```

This resulted in 5000+ deleted files.
I hope that clarifies things.

> However, I noticed that the https://github.com/googleforgames/agones/pull/3354 which you have created has over 5000 modified files.

My PR has over 5000 **deleted** files.

**Reply by Kalaiselvi84** [reply]
I've decided to consolidate everything into a single PR, so I'll close the current PRs and open a new one. Should we first remove the existing Docsy folder and then place the latest Docsy version in the same location? Additionally, it seems we need to rename hugo.toml to config.toml. Is my understanding correct?

**Reply by deining** [reply]
If you want to start from scratch again, follow the these [instructions](https://www.docsy.dev/docs/updating/convert-site-to-module/).

> Should we first remove the existing Docsy folder 

Yes.

and then place the latest Docsy version in the same location?

No. You don't need a local clone of docsy at all. 

> Additionally, it seems we need to rename hugo.toml to config.toml.

No. There is no need to rename the config file. You are _advised_ to rename it, however.

**Comment by Kalaiselvi84** [comment]
Hi, 

I have removed the existing docsy folder and I've followed the shared documentation - https://www.docsy.dev/docs/updating/convert-site-to-module/

Hugo-server is succeeded after the necessary modifications - https://gist.github.com/Kalaiselvi84/32e3948bfe66a89df2195f8989856069

Getting below error(sample included) when I executed the `make hugo-test` target - https://github.com/googleforgames/agones/blob/4d116d7d43b9cd474b1ad98bae2d308341b9ccde/build/includes/website.mk#L66

`/css/prism.css` is only present in https://github.com/googleforgames/agones/blob/4d116d7d43b9cd474b1ad98bae2d308341b9ccde/site/layouts/partials/head.html#L23

Sample error:
```
site/docs/getting-started/index.html
  target does not exist --- site/docs/getting-started/index.html --> /css/prism.css
site/docs/getting-started/create-gameserver/index.html
  target does not exist --- site/docs/getting-started/create-gameserver/index.html --> /css/prism.css
site/docs/getting-started/edit-first-gameserver-go/index.html
  target does not exist --- site/docs/getting-started/edit-first-gameserver-go/index.html --> /css/prism.css
site/docs/getting-started/create-webhook-fleetautoscaler/index.html
  target does not exist --- site/docs/getting-started/create-webhook-fleetautoscaler/index.html --> /css/prism.css
site/docs/contribute/index.html
  target does not exist --- site/docs/contribute/index.html --> /css/prism.css
site/search/index.html
  target does not exist --- site/search/index.html --> /css/prism.css
========================================================================
✘✘✘ failed in 553.903007ms
182 errors in 184 documents
```

Could you possibly provide any suggestions or leads to assist in resolving this issue?

**Comment by deining** [comment]
> Could you possibly provide any suggestions or leads to assist in resolving this issue?

You have a file [layouts/partials/head.html](https://github.com/googleforgames/agones/blob/4d116d7d43b9cd474b1ad98bae2d308341b9ccde/site/layouts/partials/head.html) inside your theme. It has 31 lines. This file overrides the docsy theme file [layouts/partials/head.html](https://github.com/google/docsy/blob/main/layouts/partials/head.html) which has 71 files.

Ergo: inside your site, you are using an outdated version of the file, this is the reason for the error you are getting. Delete your site local version of this file and you should be fine.

**Comment by Kalaiselvi84** [comment]
I have tried deleting the `site/layouts/partials/head.html` file, and this hasn't resolved the issue. 

https://gist.github.com/Kalaiselvi84/0ccffae8405ba1bc7dfc37b54874e80f

I am facing this issue when I tried to upgrade only Hugo with the latest version 0.118.2.

**Reply by Kalaiselvi84** [reply, marked as answer]
The issue was resolved by adjusting the path of the prism references in the html files.

**Comment by Kalaiselvi84** [comment]
@deining, I really/urgently need your assistance... I'm encountering alignment issues with the Agones website. Despite adjusting the HTML and CSS, the alignment doesn't appear consistent across different monitor sizes.

I followed the guide at https://www.docsy.dev/docs/updating/convert-site-to-module/, and the `hugo server` command runs without issues. All necessary targets are successful. Yet, the site's style and alignment have changed. Could you please help me address this?

**Comment by deining** [comment]
> Yet, the site's style and alignment have changed. Could you please help me address this?

Please be more specific. What files are affected? What exactly changed in these files? Can you please provide a diff between the old and the new version of affected file(s). Thanks.

**Reply by Kalaiselvi84** [reply]
PR - https://github.com/googleforgames/agones/pull/3382

Landing page alignment between Agones title/description and the player in the website is not good. You can compare the alignment between the https://f1699b2-dot-preview-dot-agones-images.appspot.com/  and https://agones.dev/site/  

Also Agones Header(left top most) in each page is not correct. Paragraph alignment for the tools in the cover page are not correct.

Some of the tabs on the header(Release and Search bar) were not working and I made adjustment in the code to make it work.

**Reply by Kalaiselvi84** [reply]
I believe I might be overlooking a detail. I attempted to use the command `hugo mod init github.com/me-at-github/my-existing-site`. Specifically, I tried variations such as `hugo mod init github.com/googleforgames/agones/tree/main/site` and `hugo mod init github.com/kalaiselvi84/agones/tree/main/site`. Reference-https://www.docsy.dev/docs/updating/convert-site-to-module/

However, I received the following error:
```
go: /usr/local/google/home/kalaiselvim/Desktop/agones/site/go.mod already exists
Error: failed to init modules: failed to execute 'go [mod init github.com/kalaiselvi84/agones/tree/main/site]': failed to execute binary "go" with args [mod init github.com/kalaiselvi84/agones/tree/main/site]: go: /usr/local/google/home/kalaiselvim/Desktop/agones/site/go.mod already exists
 *errors.errorString
```

I've ignored this error because the error message indicates that go.mod already exists and also `hugo server` was working fine. I am unsure if this could be a problem?
